charts/kuberpult/tests/charts_test.go

```diff
@@ -0,0 +1,539 @@
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_GIT_URL",
+					Value: "testURL",
+				},
+				{
+					Name:  "KUBERPULT_BOOTSTRAP_MODE",
+					Value: "true",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_PGP_KEY_RING_PATH",
+					Value: "",
+				},
+			},
+			checkValues: true,
+		},
+		{
+			Name: "Change Git URL",
+			Setup: func(t *testing.T, values *ValuesDoc) {
+				values.Git.URL = "checkThisValue"
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_GIT_URL",
+					Value: "checkThisValue",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{},
+			checkValues:     true,
+		},
+		{
+			Name: "Argo CD disabled",
+			Setup: func(t *testing.T, values *ValuesDoc) {
+				values.Git.URL = "testURL"
+				values.Argocd.GenerateFiles = false
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_ARGO_CD_GENERATE_FILES",
+					Value: "false",
+				},
+				{
+					Name:  "KUBERPULT_GIT_URL",
+					Value: "testURL",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{},
+			checkValues:     true,
+		},
+		{
+			Name: "Argo CD enabled simple test",
+			Setup: func(t *testing.T, values *ValuesDoc) {
+				values.Git.URL = "testURL"
+				values.Argocd.GenerateFiles = true
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_ARGO_CD_GENERATE_FILES",
+					Value: "true",
+				},
+				{
+					Name:  "KUBERPULT_GIT_URL",
+					Value: "testURL",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{},
+			checkValues:     true,
+		},
+		{
+			Name: "DD Metrics disabled",
+			Setup: func(t *testing.T, values *ValuesDoc) {
+				values.Git.URL = "testURL"
+				values.DatadogTracing.Enabled = false
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "KUBERPULT_GIT_URL",
+					Value: "testURL",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{
+				{
+					Name:  "DD_AGENT_HOST",
+					Value: "",
+				},
+				{
+					Name:  "DD_ENV",
+					Value: "",
+				},
+				{
+					Name:  "DD_SERVICE",
+					Value: "",
+				},
+				{
+					Name:  "DD_VERSION",
+					Value: "",
+				},
+				{
+					Name:  "KUBERPULT_ENABLE_TRACING",
+					Value: "",
+				},
+				{
+					Name:  "DD_TRACE_DEBUG",
+					Value: "",
+				},
+			},
+			checkValues: false, //Don't check the actual values of the env variables
+		},
+		{
+			Name: "DD Tracing enabled",
+			Setup: func(t *testing.T, values *ValuesDoc) {
+				values.Git.URL = "testURL"
+				values.DatadogTracing.Enabled = true
+			},
+			ExpectedEnvs: []core.EnvVar{
+				{
+					Name:  "DD_AGENT_HOST",
+					Value: "",
+				},
+				{
+					Name:  "DD_ENV",
+					Value: "",
+				},
+				{
+					Name:  "DD_SERVICE",
+					Value: "",
+				},
+				{
+					Name:  "DD_VERSION",
+					Value: "",
+				},
+				{
+					Name:  "KUBERPULT_ENABLE_TRACING",
+					Value: "",
+				},
+				{
+					Name:  "DD_TRACE_DEBUG",
+					Value: "",
+				},
+			},
+			ExpectedMissing: []core.EnvVar{},
+			checkValues:     false,
```

Comment:
I am not sure why we don't want to check the values here. I would prefer to remove this flag and always check the values.